src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -116,10 +117,12 @@ private Task StartAsyncCore()
                     if (t.IsFaulted)
                     {
                         _startTcs.SetException(t.Exception.InnerException);
+                        _closedTcs.TrySetException(t.Exception.InnerException);
                     }
                     else if (t.IsCanceled)
                     {
                         _startTcs.SetCanceled();
+                        _closedTcs.SetCanceled();
```

Comment:
if the task is faulted we use the `Try` but not here. Why? Which one should be used?

Reply:
I'm not actually completely sure here. I figure we should use Try though. @davidfowl ?

Reply:
You need to check how we complete this tcs and if the order is deterministic (I think it isn't).

Reply:
Hmm